qutebrowser/misc/throttle.py

```diff
@@ -0,0 +1,79 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2018 Jay Kamat <jaygkamat@gmail.com>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""A simple throttling decorator."""
+
+import typing
+import time
+import functools
+
+from PyQt5.QtCore import QTimer
+
+
+class throttle:  # noqa: N801,N806 pylint: disable=invalid-name
```

Comment:
You can make it a function instead to avoid the pylint warning (although I don't know if it gets more confusing)

Reply:
I structured it like this because it looks like lots of other places (see debug.py) use this same format, and I'd rather stay consistent across everything.